tests/integration/components/flash-message-test.js

```diff
@@ -40,92 +41,77 @@ test('it does not error when quickly removed from the DOM', function(assert) {
   assert.ok(this.get('flash').isDestroyed, 'Flash Object isDestroyed');
 });
 
-if (parseFloat(Ember.VERSION) > 2.0) {
-  test('flash message is removed after timeout', function(assert) {
-    assert.expect(3);
+test('flash message is removed after timeout', function(assert) {
+  assert.expect(3);
 
-    let destroyMessage = sinon.spy();
-    
-    this.set('flash', FlashMessage.create({
-      message: 'hi',
-      sticky: false,
-      timeout: timeoutDefault,
-      destroyMessage
-    }));
+  this.set('flash', FlashMessage.create({
+    message: 'hi',
+    sticky: false,
+    timeout: timeoutDefault
+  }));
 
-    this.render(hbs`
-      {{#flash-message flash=flash as |component flash|}}
-        {{flash.message}}
-      {{/flash-message}}
-    `);
+  this.render(hbs`
+    {{#flash-message flash=flash as |component flash|}}
+      {{flash.message}}
+    {{/flash-message}}
+  `);
 
-    assert.equal(this.$().text().trim(), 'hi');
-    assert.notOk(destroyMessage.calledOnce, 'flash has not been destroyed yet');
+  assert.equal(this.$().text().trim(), 'hi');
+  assert.notOk(this.get('flash').isDestroyed, 'Flash is not destroyed immediately');
 
-    return wait().then(() => {
-      assert.ok(destroyMessage.calledOnce, 'flash is destroyed after timeout');
-    });
+  return wait().then(() => {
+    assert.ok(this.get('flash').isDestroyed, 'Flash Object is destroyed');
   });
+});
 
-  test('flash message is removed after timeout', function(assert) {
-    assert.expect(3);
-
-    let destroyMessage = sinon.spy();
+test('flash message is removed after timeout if mouse enters', function(assert) {
+  assert.expect(3);
 
-    this.set('flash', FlashMessage.create({
-      message: 'hi',
-      sticky: false,
-      timeout: timeoutDefault,
-      destroyMessage
-    }));
+  let foo = FlashMessage.create({
+    message: 'hi',
+    sticky: false,
+    timeout: timeoutDefault
+  });
 
-    this.render(hbs`
-      {{#flash-message flash=flash as |component flash|}}
-        <span id="testFlash">{{flash.message}}</span>
-      {{/flash-message}}
-    `);
+  this.set('flash', foo);
 
-    assert.equal(this.$().text().trim(), 'hi');
+  this.render(hbs`
+    {{#flash-message elementId="testFlash" flash=flash as |component flash|}}
+      {{flash.message}}
+    {{/flash-message}}
+  `);
 
-    this.$('#testFlash').mouseenter();
+  assert.equal(this.$().text().trim(), 'hi');
+  this.$('#testFlash').mouseenter();
 
-    assert.notOk(
-      destroyMessage.calledOnce,
-      'flash is not destroyed after enough elapsed time'
-    );
+  assert.notOk(foo.isDestroyed, 'Flash Object is not destroyed');
+  this.$('#testFlash').mouseleave();
 
-    this.$('#testFlash').mouseleave();
-        
-    return wait().then(() => {
-      assert.ok(
-        destroyMessage.calledOnce,
-        'flash waits remaining time from original timeout'
-      );
-    });
-  });
+  later(() => {
```

Comment:
Rather than trying to be clever and use `sinon` to control the time, we'll just force it with `later`. It makes the test take a second longer to run, but we can get rid of `sinon` and this better matches the actual behavior.